Classes/ios/Scanners/MTBBarcodeScanner.m

```diff
@@ -271,7 +273,11 @@ - (void)startScanningWithResultBlock:(void (^)(NSArray *codes))resultBlock error
         if (self.didStartScanningBlock) {
             self.didStartScanningBlock();
         }
+
+        return YES;
     }
+
+    return NO;
```

Comment:
Refactor to an early `if (!session) { return NO;}` now given that the success path will become a bit more complicated when dispatching `startRunning`?  We can also refactor when doing the dispatch, though.